### ISSUE TITLE
(radar project)

### ISSUE THREAD

**Issue by VALohe** [VALohe/radar_learn#1]
from tensorflow.keras.models import load_model
import cv2
import numpy as np

def predict(image_path, model_path="models/radar_model.h5"):
    model = load_model(model_path)
    img = cv2.imread(image_path, cv2.IMREAD_GRAYSCALE)
    img = cv2.resize(img, (64, 64)).reshape(1, 64, 64, 1) / 255.0
    prediction = model.predict(img)
    return prediction.argmax(), prediction.max()
 import cv2
import numpy as np
import os

def load_images(folder, target_size=(64, 64)):
    images = []
    labels = []
    for label in os.listdir(folder):
        path = os.path.join(folder, label)
        for file in os.listdir(path):
            img = cv2.imread(os.path.join(path, file), cv2.IMREAD_GRAYSCALE)
            img = cv2.resize(img, target_size)
            images.append(img)
            labels.append(label)
    return np.array(images), np.array(labels)
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Conv2D, MaxPooling2D, Flatten, Dense
from tensorflow.keras.utils import to_categorical
from sklearn.model_selection import train_test_split
from src.preprocess import load_images

def build_model(input_shape, num_classes):
    model = Sequential([
        Conv2D(32, (3,3), activation='relu', input_shape=input_shape),
        MaxPooling2D(2,2),
        Flatten(),
        Dense(128, activation='relu'),
        Dense(num_classes, activation='softmax')
    ])
    model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['accuracy'])
    return model

def train_model(data_path):
    X, y, label_map = load_images(data_path)
    X = X.reshape(-1, 64, 64, 1) / 255.0
    y_encoded = to_categorical(y)
    X_train, X_val, y_train, y_val = train_test_split(X, y_encoded, test_size=0.2)

    model = build_model((64, 64, 1), y_encoded.shape[1])
    model.fit(X_train, y_train, epochs=10, validation_data=(X_val, y_val))
    model.save("models/radar_model.h5")

    print("Training complete. Label map:")
    for name, idx in label_map.items():
        print(f"{idx}: {name}")

from train import train_model
from predict import predict

if __name__ == "__main__":
    # Train the model
    train_model("data/")

    # Predict a sample image
    label, confidence = predict("sample_radar_image.png")
    print(f"Predicted Label: {label}, Confidence: {confidence:.2f}")
